iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/MultiplexingClientTests.java

```diff
@@ -1517,35 +1517,43 @@ public void disableDeviceAfterOpenAndAfterRegistration() throws Exception
         deviceToDisable.setStatus(DeviceStatus.Disabled);
         registryManager.updateDevice(deviceToDisable);
 
-        // verify that the disabled device loses its device session
```

Comment:
The diff here is a bit hard to follow, but I basically added a try/finally block to always re-enable the device in case it gets recycled by another test

```java
try
{
    // some test code involving a disabled device
}
finally
{
    // re-enable the device no matter what
}
```

Reply:
And I did that for the three tests in this suite that intentionally disable devices